pombola/kenya/views_facebook_experiments.py

```diff
@@ -0,0 +1,200 @@
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.views.generic import TemplateView
+from django.conf import settings
+from django.core.cache import cache
+
+from mapit.models import Area, Generation
+
+from .election_data_2017.iebc_offices import IEBC_OFFICE_DATA
+from .forms import ConstituencyGroupedByCountySelectForm
+
+import requests
+import json
+
+
+AREA_ID_TO_OFFICE = {
+    o['cons_id']: o
+    for o in IEBC_OFFICE_DATA
+}
+
+
+class TreatmentOlympics(TemplateView):
+
+    template_name = 'facebook-experiment-olympics.html'
+
+    def get(self, request, *args, **kwargs):
+        self.survey_session = request.GET.get('sid')
+
+        if self.survey_session:
+            recordPositiveResponse(self.survey_session, '62', '10173')
+
+        return super(TreatmentOlympics, self).get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super(TreatmentOlympics, self).get_context_data(**kwargs)
+        context['sid'] = self.survey_session
+        return context
+
+
+class TreatmentOlympicsThanks(TemplateView):
+
+    template_name = 'facebook-experiment-olympics-thanks.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
```

Comment:
The `sid` value is untrusted input, but it's used verbatim as part of the request to surveygizmo. At the least, I would check in each case that it's only digits (or only hexadecimals, or whatever it's meant to be!) Also, being in the URL it's very public, so it would be easy for someone to observe and repeat. Is that a problem? Are the IDs guessable?